File: LemonTracker/Models/Module.cs
namespace LemonTracker.Models;

public class Module
{
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public int InitSpeed { get; set; } = 3;
    public List<Pattern> Patterns { get; set; } = new();
    public List<Sample> Samples { get; set; } = new();
    public List<Ornament> Ornaments { get; set; } = new();
}

File: src/lib/models/module.ts
import type Ornament from './ornament';
import type Sample from './sample';
import Pattern from './pattern';

export default class Module {
	title: string;
	author: string;
	initSpeed: number;
	patterns: Pattern[];
	samples: Sample[];
	ornaments: Ornament[];

	constructor(
		title = '',
		author = '',
		initSpeed = 3,
		patterns: Pattern[] = [new Pattern(undefined, undefined, false)],
		samples: Sample[] = [],
		ornaments: Ornament[] = []
	) {
		this.title = title;
		this.author = author;
		this.initSpeed = initSpeed;
		this.patterns = patterns;
		this.samples = samples;
		this.ornaments = ornaments;
	}
}
